### PR TITLE
feat(database): implement backups 

### PR DESCRIPTION
**Pull request by KyleSanderson**
Lots of love needed, but based on the fact there's now 2k+ consumers this is becoming quite critical.

### CODE REVIEW

File: internal/database/backup.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+			retain++
+		}
+
+		return cleanupDatabase(base, db, retain)
+	}
+
+	return errors.New("backup: not implemented for database type: %s", db.Driver)
+}
+
+func databaseConsistent(tx *sql.Tx) error {
```
Comment (zze0s) [inline internal/database/backup.go:68]: Should probably make this clear it's for SQLite only, so rename it to `databaseConsistentSQLite` might be a good idea.

File: internal/database/sqlite.go [hunk 2]
```diff
@@ -78,6 +78,7 @@ func (db *DB) openSQLite() error {
 		return err
 	}
 
+	db.BackupDatabase(false)
```
Comment (zze0s) [inline internal/database/sqlite.go:81]: Do we need to run this on both startup and stop?

Comment (KyleSanderson) [reply internal/database/sqlite.go:81]: Yeah

**Comment by martylukyy** [comment]
Superseded by https://github.com/autobrr/autobrr/pull/1756

**Pull request closed without merging**